File: Tool/PreUtil/src/output/javascript/medium/CheckCompletenessOfABinaryTree.js
/**https://leetcode.com/problems/check-completeness-of-a-binary-tree/ */
//Given the root of a binary tree, determine if it is a complete binary tree.
//In a complete binary tree, every level, except possibly the last, is completely filled, and all nodes in the last level are as far left as possible. It can have between 1 and 2h nodes inclusive at the last level h.


/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isCompleteTree = function(root) {
    let q = [];
    q.push(root);
    let flag = false;
    while(q.length > 0){
        let temp = q.shift();
        if(temp == null){
            flag = true;
        }
        else{
            if(flag){
                return false;
            }
            q.push(temp.left);
            q.push(temp.right);
        }
    }
    return true;
    
};

    

File: Tool/PreUtil/src/output/python/medium/CheckCompletenessOfABinaryTree.py
#https://leetcode.com/problems/check-completeness-of-a-binary-tree/
#Given the root of a binary tree, determine if it is a complete binary tree.
#In a complete binary tree, every level, except possibly the last, is completely filled, and all nodes in the last level are as far left as possible. It can have between 1 and 2h nodes inclusive at the last level h.


# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isCompleteTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if not root:
            return True
        queue = collections.deque([root])
        while queue:
            node = queue.popleft()
            if node:
                queue.append(node.left)
                queue.append(node.right)
            else:
                break
        return not any(queue)
        

    